packages/fpsak/src/behandlingFpsak/behandlingsprosess/components/beregningsgrunnlag/beregningsgrunnlagPanel/Beregningsgrunnlag.jsx

```diff
@@ -0,0 +1,319 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import { FormattedMessage } from 'react-intl';
+import { Element } from 'nav-frontend-typografi';
+
+import {
+  getAlleAndelerIForstePeriode,
+  getBehandlingGjelderBesteberegning,
+  getBeregningsgrunnlagPerioder,
+} from 'behandlingFpsak/behandlingSelectors';
+import aksjonspunktPropType from 'behandlingFelles/proptypes/aksjonspunktPropType';
+import BehandlingspunktSubmitButton from 'behandlingFpsak/behandlingsprosess/components/BehandlingspunktSubmitButton';
+import aksjonspunktCodes from '@fpsak-frontend/kodeverk/src/aksjonspunktCodes';
+import { BorderBox, ElementWrapper, VerticalSpacer } from '@fpsak-frontend/shared-components';
+import periodeAarsak from '@fpsak-frontend/kodeverk/src/periodeAarsak';
+import { TextAreaField } from '@fpsak-frontend/form';
+import {
+  hasValidText, maxLength, minLength, required, removeSpacesFromNumber,
+} from '@fpsak-frontend/utils';
+import YtelserFraInfotrygd
+  from 'behandlingFpsak/behandlingsprosess/components/beregningsgrunnlag/tilstotendeYtelser/YtelserFraInfotrygd';
+import GrunnlagForAarsinntektPanelFL from '../frilanser/GrunnlagForAarsinntektPanelFL';
+import GrunnlagForAarsinntektPanelAT from '../arbeidstaker/GrunnlagForAarsinntektPanelAT';
+import GrunnlagForAarsinntektPanelSN from '../selvstendigNaeringsdrivende/GrunnlagForAarsinntektPanelSN';
+import FastsettNaeringsinntektSN from '../selvstendigNaeringsdrivende/FastsettNaeringsinntektSN';
+import OppsummeringSN from '../selvstendigNaeringsdrivende/OppsummeringSN';
+import TilstotendeYtelser from '../tilstotendeYtelser/TilstotendeYtelser';
+import MilitaerPanel from '../militær/MilitaerPanel';
+import FastsettInntektTidsbegrenset from '../arbeidstaker/FastsettInntektTidsbegrenset';
+import styles from './beregningsgrunnlag.less';
+
+// ------------------------------------------------------------------------------------------ //
+// Variables
+// ------------------------------------------------------------------------------------------ //
+
+export const TEKSTFELTNAVN_BEGRUNN_DEKNINGSGRAD_ENDRING = 'begrunnDekningsgradEndring';
+
+const {
+  FASTSETT_BEREGNINGSGRUNNLAG_ARBEIDSTAKER_FRILANS,
+  VURDER_VARIG_ENDRET_ELLER_NYOPPSTARTET_NAERING_SELVSTENDIG_NAERINGSDRIVENDE,
+  FASTSETT_BEREGNINGSGRUNNLAG_TIDSBEGRENSET_ARBEIDSFORHOLD,
+  FASTSETT_BEREGNINGSGRUNNLAG_SN_NY_I_ARBEIDSLIVET,
+  VURDER_DEKNINGSGRAD,
+} = aksjonspunktCodes;
+const minLength3 = minLength(3);
+const maxLength1500 = maxLength(1500);
+
+// ------------------------------------------------------------------------------------------ //
+// Methods
+// ------------------------------------------------------------------------------------------ //
+
+const harPerioderMedAvsluttedeArbeidsforhold = (allePerioder, gjeldendeAksjonspunkter) => allePerioder.filter(periode => periode.periodeAarsaker !== null
```

Comment:
Ville kanskje skrevet dette som
```suggestion
const harPerioderMedAvsluttedeArbeidsforhold = (allePerioder, gjeldendeAksjonspunkter) => allePerioder
.some(({ periodeAarsaker }) => periodeAarsaker
  && periodeAarsaker.some(({ kode }) => kode === periodeAarsak.ARBEIDSFORHOLD_AVSLUTTET)))
  && (gjeldendeAksjonspunkter.some(ap => ap.definisjon.kode === FASTSETT_BEREGNINGSGRUNNLAG_TIDSBEGRENSET_ARBEIDSFORHOLD));
```

Reply:
Lurt! Skrev den sånn.